src/dataflow/src/sink/kafka.rs

```diff
@@ -394,10 +397,40 @@ impl KafkaSinkState {
         }
     }
 
-    fn send<K, P>(&self, record: BaseRecord<K, P>) -> Result<(), bool>
+    async fn init_transactions(&self, timeout: Duration) -> KafkaResult<()> {
+        let self_producer = self.producer.clone();
+        task::spawn_blocking(move || self_producer.init_transactions(timeout))
+            .await
+            .unwrap_or(Err(KafkaError::Canceled))
+    }
+
+    async fn begin_transaction(&self) -> KafkaResult<()> {
+        let self_producer = self.producer.clone();
+        task::spawn_blocking(move || self_producer.begin_transaction())
+            .await
+            .unwrap_or(Err(KafkaError::Canceled))
+    }
+
+    async fn commit_transaction(&self, timeout: Duration) -> KafkaResult<()> {
+        let self_producer = self.producer.clone();
+        task::spawn_blocking(move || self_producer.commit_transaction(timeout))
+            .await
+            .unwrap_or(Err(KafkaError::Canceled))
+    }
+
+    async fn abort_transaction(&self, timeout: Duration) -> KafkaResult<()> {
+        let self_producer = self.producer.clone();
+        task::spawn_blocking(move || self_producer.abort_transaction(timeout))
+            .await
+            .unwrap_or(Err(KafkaError::Canceled))
+    }
+
+    // TODO: make calls to ThreadedProducer::send on a backgroundthread.  Currently BaseRecord
```

Comment:
`send()` will never block because it just enqueues, so it should be fine to not make this one `async`.

Reply:
Any errors will be presented to `ProducerContext::delivery()`

Reply:
Ahh thank you for letting me know! I'll update